### ISSUE TITLE
Add support for transitive dependencies to expiring-todo-comments

### ISSUE THREAD

**Issue by strangedev** [sindresorhus/eslint-plugin-unicorn#1135]
Hi there :wave: 
This is more of a feature request than a bug report, as the README already states that this functionality does not currently exist.
Basically, when we pull in a dependency transitively, e.g. pulling in TypeScript through [roboter](https://github.com/thenativeweb/roboter), it'd be nice to be able to check not only the version of the dependency itself, but also the version of the transitive dependency.

This would be especially helpful for the mentioned use-case, as TypeScript sometimes breaks the API in minor versions and we'd like to tag certain lines of code to be reconsidered once a new TypeScript version has landed in roboter.

I've prepared a small example of the "issue" here: https://github.com/strangedev/unicorn-expiring-todo-comments

Basically, we pull in roboter at some version and want to check that the included TypeScript version is above some specific version.
```js
// TODO [typescript@>=4.1.5]: This transitive dependency is outdated.
// TODO [roboter@>=10.0.0]: This dependency is outdated.
console.log('Ohai! :3');
```

It'd be nice if the rule would fail for both lines and not just the one where the dependency is explicitly listed in `package.json`.

Anyway, thanks for the great work!

**Comment by strangedev** [comment]
@yeldiRium and I had a look at the code and thought about this feature a little bit. It seems we can't really decide which version of a transitive dependency should be considered, since there may be several versions installed through different dependencies.

Could be that this feature is not really possible after all. If anyone's got ideas on this though, we'd be happy to hear / discuss them :smile:

**Comment by fisker** [comment]
[`resolutions` field in `yarn` uses `/`](https://github.com/yarnpkg/rfcs/blob/master/implemented/0000-selective-versions-resolutions.md#package-designation)

**Comment by sindresorhus** [comment]
I do not think this belongs in this rule.

`expiring-todo-comments` intentionally checks stable project metadata from the nearest `package.json`: package version, engines, direct dependencies/devDependencies, and peer dependency floors. Transitive dependencies are different. They are part of the package manager's resolved install graph, and the same package can appear multiple times at different versions through different paths.

Supporting this properly would mean defining path syntax, reading package-manager-specific lockfiles or `node_modules`, handling npm/Yarn/pnpm/workspace differences, and deciding what to do when several matching versions exist. That is a lot of complexity for a TODO rule, and it would still be easy to misunderstand.

If a transitive dependency matters enough to gate source cleanup, I think the better options are to declare it directly, use package manager overrides/resolutions, or track it with dependency tooling.
